### PR TITLE
Cancel dashboard requests

### LINKED ISSUE #1846: [Dashboard] Cancel older in flight http requests when filters are changed
We are not cancelling older queries when filters are changed. When run locally http queue takes care of not sending older queries. But in deployed context since the requests are in flight those need to be cancelled.

### PR DESCRIPTION
**Pull request by AdityaHegde**
closes #1846

### CODE REVIEW

**Comment by ericpgreen2** [comment]
A couple general comments:
- Shouldn’t `cancelDashboardQueries(...)` go immediately before we change dashboard state, not after? If after, I worry we’ll cancel the requests we just issued.
- Did you consider putting the cancelation calls in `dashboard-stores.ts` rather than in the component code? Thoughts on that approach?

File: web-common/src/runtime-client/http-request-queue/HttpRequestQueue.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ try {
   }
 } catch (err) {
   // no-op
```
Comment (ericpgreen2) [inline web-common/src/runtime-client/http-request-queue/HttpRequestQueue.ts:36]: No longer a no-op. It also looks like this code could be simplied. Does this mean the queue should always be 200?

File: web-local/src/lib/svelte-query/invalidation.ts [hunk 2]
```diff
@@ -68,12 +68,20 @@ const getInvalidationsForPath = (
   }
 };
 
-export function invalidationForMetricsViewData(query, metricsViewName: string) {
+export function invalidationForMetricsViewDataKey(
+  queryHash,
```
Comment (ericpgreen2) [inline web-local/src/lib/svelte-query/invalidation.ts:72]: Missing type

File: web-local/src/lib/svelte-query/invalidation.ts [hunk 3]
```diff
@@ -68,12 +68,20 @@ const getInvalidationsForPath = (
...
+  metricsViewName: string
+) {
   const r = new RegExp(
     `/v1/instances/[a-zA-Z0-9-]+/queries/metrics-views/${metricsViewName}/`
   );
-
-  return typeof query.queryKey[0] === "string" && r.test(query.queryKey[0]);
+  return r.test(queryHash);
+}
+export function invalidationForMetricsViewData(query, metricsViewName: string) {
```
Comment (ericpgreen2) [inline web-local/src/lib/svelte-query/invalidation.ts:80]: Missing type

File: web-local/src/lib/svelte-query/invalidation.ts [hunk 4]
```diff
@@ -68,12 +68,20 @@ const getInvalidationsForPath = (
   }
 };
 
-export function invalidationForMetricsViewData(query, metricsViewName: string) {
+export function invalidationForMetricsViewDataKey(
```
Comment (ericpgreen2) [inline web-local/src/lib/svelte-query/invalidation.ts:71]: I find these "invalidationXXX" function names a bit confusing. What about `isMetricsViewQuery`?

File: web-common/src/runtime-client/http-request-queue/HttpRequestQueue.ts (hunk 1 above)
Comment (AdityaHegde) [reply web-common/src/runtime-client/http-request-queue/HttpRequestQueue.ts:36]: This was for testing. Will revert.

File: web-local/src/lib/svelte-query/invalidation.ts (hunk 3 above)
Comment (AdityaHegde) [reply web-local/src/lib/svelte-query/invalidation.ts:80]: Not worth writing explicit query type here.

**Comment by AdityaHegde** [comment]
Updated `cancelDashboardQueries` to run before store is updated.

Adding query related stuff in store doesnt feel right.

File: web-common/src/features/dashboards/dashboard-queries.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import type { QueryClient } from "@sveltestack/svelte-query";
+
+export function cancelDashboardQueries(
+  queryClient: QueryClient,
+  metricsViewName: string
+) {
+  return queryClient.cancelQueries({
+    fetching: true,
+    predicate: (query) => {
+      return isMetricsViewData(query.queryHash, metricsViewName);
```
Comment (ericpgreen2) [inline web-common/src/features/dashboards/dashboard-queries.ts:11]: `isMetricsViewQuery`?

**Pull request merged**